pkg/controllers/v1alpha2/mpi_job_controller.go

```diff
@@ -530,15 +530,14 @@ func (c *MPIJobController) syncHandler(key string) error {
 		if err != nil {
 			return err
 		}
-
-		// If the worker is ready, start the launcher.
-		workerReady := worker.Status.ReadyReplicas == workerReplicas
-		if workerReady && launcher == nil {
+		_, lancherErr := c.kubeClient.BatchV1().Jobs(namespace).Get(mpiJob.Name+launcherSuffix, metav1.GetOptions{})
```

Comment:
We do not need get mpijob from kube-apiserver again, we have just get launcher above from the lister, so just judge whether the launcher is nil.

Reply:
got it